src/user-statuses/UserStatusScreen.js

```diff
@@ -36,74 +74,137 @@ type Props = $ReadOnly<{|
   route: RouteProp<'user-status', void>,
 |}>;
 
+const statusTextFromInputValue = (v: string): $PropertyType<UserStatus, 'status_text'> =>
+  v.trim() || null;
+
+const inputValueFromStatusText = (t: $PropertyType<UserStatus, 'status_text'>): string => t ?? '';
+
+const statusEmojiFromInputValue = (v: EmojiInputValue): $PropertyType<UserStatus, 'status_emoji'> =>
+  v
+    ? {
+        emoji_name: v.name,
+        emoji_code: v.code,
+        reaction_type: reactionTypeFromEmojiType(v.type, v.name),
+      }
+    : null;
+
+const inputValueFromStatusEmoji = (e: $PropertyType<UserStatus, 'status_emoji'>): EmojiInputValue =>
+  e
+    ? {
+        type: emojiTypeFromReactionType(e.reaction_type),
+        code: e.emoji_code,
+        name: e.emoji_name,
+      }
+    : null;
+
 export default function UserStatusScreen(props: Props): Node {
   const { navigation } = props;
+  const _ = useContext(TranslationContext);
```

Comment:
UserStatusScreen [nfc]: Move `useContext` upward, with props
    
    As Greg suggests:
      https://github.com/zulip/zulip-mobile/pull/5277#discussion_r820978534

Just to be sure you saw: my suggestion was actually to put it with the selectors, not with the props. But here with the props is fine too.

Reply:
Ah, indeed! I've gone and put it with the selectors for the next revision.